.github/workflows/apiBreakTest.yml

```diff
@@ -0,0 +1,174 @@
+name: "Check API Breakage"
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  buildAndCheckAPIBreakage:
+    name: Build and Check API Breakage
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        fetch-depth: 0
+
+
+    - name: Setup and Run Swift API Diff
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        echo "Swift version: $(swift --version)"
+        echo "Swift package manager version: $(swift package --version)"
+        swift package resolve
+
+        # Ensure we are in the correct directory
+        cd $GITHUB_WORKSPACE
+        
+        # Run swift-api-diff commands here directly
+        NEW_API_DIR=$(mktemp -d)
+        OLD_API_DIR=$(mktemp -d)
+        SDK_PATH=$(xcrun --show-sdk-path)
+
+        # Get all library module names
```

Comment:
I thought the design has been updated so that the old.json doesn't need to be built and generated every time, it looks like this is still checking out and generating the old json files.

Reply:
This is a backup, if github action cannot find any api files in the base branch, then it will generate the old json files for it.